static/app/components/core/avatar/index.mdx

```diff
@@ -0,0 +1,318 @@
+  },
+  author: 'John Doe',
+  description: 'Test Doc Integration',
+  isDraft: false,
+  popularity: 0,
+  url: 'https://sentry.io/test-doc-integration',
+};
+
+There are multiple avatar components which represent users, teams, organizations, projects, and integrations.
+
+<Storybook.Demo>
+  <UserAvatar size={PREVIEW_SIZE} user={USER} />
+  <TeamAvatar size={PREVIEW_SIZE} team={TEAM} />
+  <ProjectAvatar size={PREVIEW_SIZE} project={PROJECT} />
+  <OrganizationAvatar size={PREVIEW_SIZE} organization={ORGANIZATION} />
+  <SentryAppAvatar size={PREVIEW_SIZE} sentryApp={SENTRY_APP} />
+  <DocIntegrationAvatar size={PREVIEW_SIZE} docIntegration={DOC_INTEGRATION} />
+</Storybook.Demo>
+
+## Props
+
+All avatar components accept common props, to customize `size` and render tooltips with `hasTooltip`, `tooltip`, and `tooltipOptions`.
+
+<Storybook.Demo>
+  <UserAvatar user={USER} size={64} hasTooltip tooltip="This avatar has a tooltip" />
+</Storybook.Demo>
+```jsx
+<UserAvatar user={user} size={64} hasTooltip tooltip="This avatar has a tooltip" />
+```
+
+## Types
+
+To distinguish between individuals (users) and groups (teams, organizations, etc) at glance, avatars may use different shapes. Individuals are displayed with a round avatar, but groups are displayed with a square avatar.
+
+### User
+
+The `<UserAvatar />` component displays an avatar for a user.
+
+<Storybook.Demo>
+  <UserAvatar size={PREVIEW_SIZE} user={USER} />
+</Storybook.Demo>
+```jsx
+<UserAvatar user={user} />
+```
+
+### Team
+
+The `<TeamAvatar />` component displays an avatar for a team.
+
+<Storybook.Demo>
+  <TeamAvatar size={PREVIEW_SIZE} team={TEAM} />
+</Storybook.Demo>
+```jsx
+<TeamAvatar team={team} />
+```
+
+### Project
+
+The `<ProjectAvatar project={project} />` component displays an avatar for a project.
+
+<Storybook.Demo>
+  <ProjectAvatar size={PREVIEW_SIZE} project={PROJECT} />
+</Storybook.Demo>
+```jsx
+<ProjectAvatar project={project} />
+```
+
+### Organization
+
+The `<OrganizationAvatar />` component displays an avatar for an organization.
+
+<Storybook.Demo>
+  <OrganizationAvatar size={PREVIEW_SIZE} organization={ORGANIZATION} />
+</Storybook.Demo>
+```jsx
+<OrganizationAvatar organization={organization} />
+```
+
+### Sentry App
+
+The `<SentryAppAvatar sentryApp={sentryApp} />` component displays an avatar for a SentryApp (integration).
+
+<Storybook.Demo>
+  <SentryAppAvatar size={PREVIEW_SIZE} sentryApp={SENTRY_APP} />
+</Storybook.Demo>
+```jsx
+<SentryAppAvatar sentryApp={sentryApp} />
+```
+
+### Doc Integration
+
+The `<DocIntegrationAvatar />` component displays an avatar for a doc integration.
+
+<Storybook.Demo>
+  <DocIntegrationAvatar size={PREVIEW_SIZE} docIntegration={DOC_INTEGRATION} />
+</Storybook.Demo>
+```jsx
+<DocIntegrationAvatar docIntegration={docIntegration} />
+```
+
+## Avatar Lists
+
+To display multiple avatars in a group, use the `AvatarList` component. It accepts `users` or `teams`.
```

Comment:
just thinking out loud, for consistency: should this be `Avatar.List` ? I’m not yet quite following when we use the dot notation and when not. Or is the dot notation only used when components should be used / composed together?

Reply:
It's a great question! I don't think we have any consistent rule of thumb here yet and everything is pretty case-by-case. Seems like a great opportunity for us to standardize!